fieldsolver/gridGlue.cpp

```diff
@@ -92,6 +92,118 @@ template <typename T, int stencil> void computeCoupling(dccrg::Dccrg<SpatialCell
   }
 }
 
+/*
+Filter moments after feeding them to FsGrid to alleviate the staircase effect caused in AMR runs.
+This is using a 3D, 5-point stencil triangle kernel.
+*/
+void filterMoments(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+                           FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH> & momentsGrid,
+                           FsGrid< fsgrids::technical, FS_STENCIL_WIDTH> & technicalGrid) 
+{
+
+
+
+   // Kernel Characteristics
+   const int stencilWidth = 5;   // dimension size of a 3D kernel
+   const int kernelOffset = 2;   // offset of 5 pointstencil 3D kernel => (floor(stencilWidth/2);)
+   const Real kernelSum=729.0;
+   const Real kernel[5][5][5] ={
+                                 {{ 1,  2,  3,  2,  1},
+                                 { 2,  4,  6,  4,  2},
+                                 { 3,  6,  9,  6,  3},
+                                 { 2,  4,  6,  4,  2},
+                                 { 1,  2,  3,  2,  1}},
+
+                                 {{ 2,  4,  6,  4,  2},
+                                 { 4,  8, 12,  8,  4},
+                                 { 6, 12, 18, 12,  6},
+                                 { 4,  8, 12,  8,  4},
+                                 { 2,  4,  6,  4,  2}},
+
+                                 {{ 3,  6,  9,  6,  3},
+                                 { 6, 12, 18, 12,  6},
+                                 { 9, 18, 27, 18,  9},
+                                 { 6, 12, 18, 12,  6},
+                                 { 3,  6,  9,  6,  3}},
+
+                                 {{ 2,  4,  6,  4,  2},
+                                 { 4,  8, 12,  8,  4},
+                                 { 6, 12, 18, 12,  6},
+                                 { 4,  8, 12,  8,  4},
+                                 { 2,  4,  6,  4,  2}},
+
+                                 {{ 1,  2,  3,  2,  1},
+                                 { 2,  4,  6,  4,  2},
+                                 { 3,  6,  9,  6,  3},
+                                 { 2,  4,  6,  4,  2},
+                                 { 1,  2,  3,  2,  1}}
+                                 };
```

Comment:
I would say this is a textbook example for consteval, if I have ever seen one. :)

However, I appreciate that this code is readable, whereas a consteval function probably wouldn't.

Anyway, my suggestion: make this ```const static```. Not sure if it makes a big difference performance-wise though.

Reply:
Consteval is a c++20 feauture I think. Will make it a const static then, thanks!